### PR TITLE
Feature/wod 2 embed web ordering journey

### PR DESCRIPTION
**Pull request by LokmaneKrizou**
Screenshot 2021-08-20 at 17 57 36

### CODE REVIEW

File: app/src/main/java/com/realifetech/sample/MainActivity.kt [hunk 1]
```diff
@@ -29,12 +31,16 @@ class MainActivity : AppCompatActivity() {
             RealifeTech.getGeneral().configuration.graphApiUrl = text.toString()
             storage.graphQl = text.toString()
         }
-
+        orderingUrlEditTextView.doOnTextChanged { text, _, _, _ ->
```
Comment (sjulia7) [inline app/src/main/java/com/realifetech/sample/MainActivity.kt:34]: Can we extract this couple of lines to a separate function?

File: app/src/main/java/com/realifetech/sample/SampleApplication.kt [hunk 2]
```diff
@@ -23,5 +26,11 @@ class SampleApplication : Application() {
             clientSecret = storage.clientSecret
             appCode = storage.clientId
         }
+//  Set Colors via code
```
Comment (sjulia7) [inline app/src/main/java/com/realifetech/sample/SampleApplication.kt:29]: It this needed?

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFeature.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.realifetech.sdk.sell.weboredering
+
+import android.content.Intent
+import com.realifetech.core_sdk.domain.CoreConfiguration.context
+
+object WebOrderingFeature {
```
Comment (sjulia7) [inline sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFeature.kt:6]: <3

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                setOnClickListener { webView.goBack() }
+            }
+        }
+    }
+
+    private fun setupToolbar() {
+        binding.apply {
+            val activity = activity as? AppCompatActivity
+            activity?.setSupportActionBar(toolbar)
+            activity?.supportActionBar?.apply {
```
Comment (sjulia7) [inline sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt:64]: I guess the lines below are going to be used in several places with the time. Can we already create an extension function, similar to setUpToolbarWithBackButton / setUpToolbar in our platform app?

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    private fun setupWebView() {
+        binding.apply {
+            webView.apply {
+                context?.let {
+                    setBackgroundColor(ContextCompat.getColor(it, R.color.RLT_SDK_Color_Surface))
+                }
+                setOnLongClickListener { true }
+                isHapticFeedbackEnabled = false
+                isLongClickable = false
+                settings.apply {
```
Comment (sjulia7) [inline sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt:88]: Can we split this function into smaller ones as well? Like a different one to setupSettings() for example that is going to wrap up lines between 88 - 100?

File: sdk/src/main/java/com/realifetech/sdk/utils/ColorPallet.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package com.realifetech.sdk.utils
+
+import androidx.annotation.ColorInt
+import androidx.core.content.ContextCompat
+import com.realifetech.core_sdk.domain.CoreConfiguration
+import com.realifetech.realifetech_sdk.R
+
+// require context, should be tested using UI test
+internal object ColorPallet {
```
Comment (sjulia7) [inline sdk/src/main/java/com/realifetech/sdk/utils/ColorPallet.kt:9]: <3

**Review by sjulia7 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/realifetech/sample/MainActivity.kt [hunk 7]
```diff
@@ -29,12 +31,16 @@ class MainActivity : AppCompatActivity() {
...
+            RealifeTech.getGeneral().configuration.webOrderingJourneyUrl = text.toString()
```
Comment (mickeylee25) [inline app/src/main/java/com/realifetech/sample/MainActivity.kt:35]: Okay, so I will also move  `webOrderingJourneyUrl` to the configuration level in next iOS PR.

File: app/src/main/java/com/realifetech/sample/MainActivity.kt [hunk 8]
```diff
@@ -29,12 +31,16 @@ class MainActivity : AppCompatActivity() {
...
+            storage.orderingJourney = text.toString()
+        }
         generalButton.setOnClickListener { GeneralSampleActivity.start(this) }
         communicateButton.setOnClickListener { CommunicationSampleActivity.start(this) }
         analyticsButton.setOnClickListener { AnalyticsSampleActivity.start(this) }
         audienceButton.setOnClickListener { AudienceSampleActivity.start(this) }
         widgetsButton.setOnClickListener { WidgetsSampleActivity.start(this) }
         webPage.setOnClickListener { WebPageSampleActivity.start(this) }
+        ordering_journey.setOnClickListener { Sell.createOrderingJourneyViewController() }
```
Comment (mickeylee25) [inline app/src/main/java/com/realifetech/sample/MainActivity.kt:44]: I thought Android would call it `createOrderingJourneyFragment`

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+                    if (URLUtil.isNetworkUrl(url)) {
+                        return false
+                    }
+                }
+                return true
+            }
+
+            override fun onLoadResource(view: WebView, url: String) {
+                webView.settings.cacheMode =
+                    if (isNetworkAvailable()) LOAD_DEFAULT else LOAD_CACHE_ELSE_NETWORK
```
Comment (mickeylee25) [inline sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt:141]: Okay, I will use the similar cache policy for iOS

**Review by Fabio-Insolia-LiveStyled — APPROVED** [review]

File: app/src/main/java/com/realifetech/sample/SampleApplication.kt (hunk 2 above)
Comment (LokmaneKrizou) [reply app/src/main/java/com/realifetech/sample/SampleApplication.kt:29]: it shows example of how you can use colors via code

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt (hunk 4 above)
Comment (LokmaneKrizou) [reply sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt:64]: not really, at this stage we have only one view, and i am not aware if there will be more features requires views, when that time comes, i will make an extension for this one

File: app/src/main/java/com/realifetech/sample/MainActivity.kt (hunk 7 above)
Comment (LokmaneKrizou) [reply app/src/main/java/com/realifetech/sample/MainActivity.kt:35]: where do you have it right now?

File: app/src/main/java/com/realifetech/sample/MainActivity.kt (hunk 8 above)
Comment (LokmaneKrizou) [reply app/src/main/java/com/realifetech/sample/MainActivity.kt:44]: yeah, that was written in the document, but you are right, i ll change it to fragment, it does make more sense.

File: app/src/main/java/com/realifetech/sample/MainActivity.kt (hunk 7 above)
Comment (mickeylee25) [reply app/src/main/java/com/realifetech/sample/MainActivity.kt:35]: I set the URL with `RealifeTech().set(webOrderingJourneyUrl: String)` according to the document, but adding ` webOrderingJourneyUrl` to the SDK configuration and is assigned to the Sell when configuring the SDK makes more sense for me.

Comment (LokmaneKrizou) [reply app/src/main/java/com/realifetech/sample/MainActivity.kt:35]: i see your point, Let's agree then to use Realifetech.getSell().configuration. webOrderingJourneyUrl = some text

File: sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt (hunk 4 above)
Comment (sjulia7) [reply sdk/src/main/java/com/realifetech/sdk/sell/weboredering/WebOrderingFragment.kt:83]: ok I see, thanks!

**Review by sjulia7 — APPROVED** [review]

File: app/src/main/java/com/realifetech/sample/MainActivity.kt (hunk 7 above)
Comment (mickeylee25) [reply app/src/main/java/com/realifetech/sample/MainActivity.kt:46]: Okay, I've created a function `RealifeTech().set(webOrderingJourneyUrl: String)` to match architecture though. To conclude, there are two ways of setting the web ordering journey URL to the SDK.
1. Through SDK configuration when the host app initialises the SDK.
2. Override the URL via `set(webOrderingJourneyUrl: String)` interface from the host app if they want. -> We will add the READEME as the [architecture](https://livestyled.atlassian.net/wiki/spaces/PROD/pages/2966487103/Mobile+SDK+-+Web+Ordering+Journey+Mobile+Architecture#Update-Documentation) says.

**Review by mickeylee25 — APPROVED** [review]

**Review by Fabio-Insolia-LiveStyled — APPROVED** [review]

**Pull request merged**